File: ch3/3_5rotation_str.py
# -*- coding:utf-8 -*-

class Rotation:
    def chkRotation(self, A, lena, B, lenb):
        # write code here
        result = False
        if lenb != lena:
            pass
        else:
            str_cat = A + A
            print str_cat
            for idx in xrange(lena):
                if str_cat[idx : idx+lena] == B:
                    result = True
        return result

File: 3_string/3_5rotation_str.cpp
//#include<string>
//using namespace std;
class Rotation {
    public:
        bool chkRotation(string A, int lena, string B, int lenb) {
            // write code here
            bool result = false;
            if ( lena != lenb){
                return result;
            }
            string sum = A + A;
            for(int i = 0; i < lena; i++){
                if(sum.substr(i, lena) == B){
                    result = true;
                }
            }
        return result;
        }
};
